infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/openapi/request/OneOfJsonRequestContentTypeDefinition.java

```diff
@@ -44,10 +44,6 @@ public T deserialize(final InputStream in) throws IOException {
       throw new MissingRequestBodyException();
     }
 
-    if (!openApiTypeDefinition.isEquivalentToDeserializableType(type)) {
-      throw new IllegalStateException(
-          "Schema determined for parsing request body is not listed in requestBodyTypes");
-    }
```

Comment:
this seems like a valid constraint, why are we removing it?

Reply:
Put it back and refactored `isEquivalentToDeserializableType` slightly